documentation/workflow/run-on-server-eclipse.asciidoc

```diff
@@ -0,0 +1,41 @@
+---
+title: Run on Server in Eclipse
+order: 8
+layout: page
+---
+
+= Run on Server in Eclipse
+
+You can run and debug your Vaadin application in Eclipse on the server of your choice (such as Tomcat or Wildfly).
+You just need to install the corresponding application server plugin via the Eclipse Marketplace, then follow the plugin's instructions for setting up a development server and deploying your projects.
+After successful installation and configuration, the configured server will be visible in the "Servers" pane.
+Via this pane you start/stop the server and deploy/remove workspace modules.
+
+The `prepare-frontend` goal of the Vaadin Maven plugin should be run in the project directory and is responsible for generating a token file including among other things the path to the project directory when running in development mode (required for Webpack and npm/pnpm).
+This file is called  `flow-build-info.json` file and must be included in the WAR file before deployment.
+In normal cases when using `m2e` (Eclipse Maven Integration), this happens automatically on project configure / deploy as the Vaadin Maven plugin embeds the following lifecycle control metadata:
+
+```xml
+<lifecycleMappingMetadata>
+  <pluginExecutions>
+    <pluginExecution>
+      <pluginExecutionFilter>
+        <goals>
+          <goal>prepare-frontend</goal>
+        </goals>
+      </pluginExecutionFilter>
+      <action>
+        <execute>
+          <runOnIncremental>false</runOnIncremental>
+          <runOnConfiguration>true</runOnConfiguration>
+        </execute>
+      </action>
+    </pluginExecution>
+  </pluginExecutions>
+</lifecycleMappingMetadata>
+```
+
+== Running `prepare-frontend` manually
+If for some reason `prepare-frontend` is not triggered automatically, the WAR may be created without `flow-build-info.json`.
+In this case you will see an exception during server startup when trying to initialize development mode.
```

Comment:
It would be helpful to place here an example of the exception thrown because of `flow-build-info.json` lack. Just to help the developers to recognise this case versus to other cases when they get exceptions for other reasons.

Reply:
Good point, done.